trieste/acquisition/function/multi_objective.py

```diff
@@ -379,3 +384,245 @@ def _update_expected_improvement_fn(
                 isinstance(self._expected_improvement_fn, expected_hv_improvement), []
             )
             self._expected_improvement_fn.update(_partition_bounds)  # type: ignore
+
+
+class HippoPenalizationAcquisitionFunction(GreedyAcquisitionFunctionBuilder[ProbabilisticModel]):
```

Comment:
Is there anyway of inheriting from the local penalization builder? It looks very similar to yours

Reply:
Not directly, because it has some bits we don't want, like penalizer choice (we don't have a choice) and sampling (which we don't do). But we could maybe consider moving their common bits out?

Reply:
@uri-granta ?